### DISCUSSION TITLE
ISR shared between deployments

### DISCUSSION THREAD

**Discussion by andrewdoro** [vercel/community#3356] [Feature Question] [answered]
### Question

I've found that you can do a instant rollback but, I couldn't find any information related to the ISR cached being synced between deployments. I am not sure if the cache is being synced between environments or it only works for production. It would be helpful if anyone could clarify how this is working
https://vercel.com/docs/concepts/incremental-static-regeneration

**Comment by leerob** [comment, marked as answer]
> Instant Rollbacks: ISR allows you to roll back instantly and not lose your previously generated pages by persisting them between deployments

If you have a production deployment where X pages are generated and cached, then a new deployment Y where other pages are generated and cached, the values are keyed based on the deployment. When you do a rollback, you are now "pointing back" to deployment X, thus you can read the pages you've already generated again which needing to re-run `getStaticProps`.
